opensearchapi/api_search-params.go

```diff
@@ -0,0 +1,284 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+//
+// Modifications Copyright OpenSearch Contributors. See
+// GitHub history for details.
+
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package opensearchapi
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// SearchParams represents possible parameters for the SearchReq
+type SearchParams struct {
+	AllowNoIndices             *bool
+	AllowPartialSearchResults  *bool
+	Analyzer                   string
+	AnalyzeWildcard            *bool
+	BatchedReduceSize          *int
+	CcsMinimizeRoundtrips      *bool
+	DefaultOperator            string
+	Df                         string
+	DocvalueFields             []string
+	ExpandWildcards            string
+	Explain                    *bool
+	From                       *int
+	IgnoreThrottled            *bool
+	IgnoreUnavailable          *bool
+	Lenient                    *bool
+	MaxConcurrentShardRequests *int
+	MinCompatibleShardNode     string
+	Preference                 string
+	PreFilterShardSize         *int
+	Query                      string
```

Comment:
It would be great if Query can also be defined as a struct instead of leaving it as a string. One possible inspiration could be this one: https://github.com/mottaquikarim/esquerydsl/blob/master/esquerydsl.go

Reply:
That is not my goal for this PR. What you want would be more related to https://github.com/opensearch-project/opensearch-go/issues/184 otherwise feel free to open an issue for your request.